internal/auth/auth.go

```diff
@@ -43,30 +43,27 @@ func getUniverseDomain(ctx context.Context, contents []byte, scope string) (stri
 }
 
 // Create token source from the JSON file at the supplide path.
-func newTokenSourceFromPath(
-	ctx context.Context,
-	path string,
-	scope string,
-) (ts oauth2.TokenSource, err error) {
+func newTokenSourceFromPath(ctx context.Context, path string, scope string) (oauth2.TokenSource, error) {
 	// Read the file.
 	contents, err := os.ReadFile(path)
 	if err != nil {
 		err = fmt.Errorf("ReadFile(%q): %w", path, err)
-		return
+		return nil, err
 	}
 
 	// By default, a standard OAuth 2.0 token source is created
 	// Create a config struct based on its contents.
 	jwtConfig, err := google.JWTConfigFromJSON(contents, scope)
 	if err != nil {
 		err = fmt.Errorf("JWTConfigFromJSON: %w", err)
+		return nil, err
 	}
 	// Create the token source.
-	ts = jwtConfig.TokenSource(ctx)
+	ts := jwtConfig.TokenSource(ctx)
```

Comment:
If we are not using `ts`, why are we creating here?

Reply:
We are using right. If it will not go in the if condition then it will return ts and err at the end.